tests/tools/test_clang.lua

```diff
@@ -48,6 +48,16 @@
 		test.isequal("windres-16", clang.gettoolname(cfg, "rc"))
 	end
 
+--
+-- Verify that toolsetpath overrides the default tool name.
+--
+function suite.toolsetpathOverridesDefault()
+	toolset "clang"
+	toolsetpath("clang", "cc", "/path/to/my/custom/clang")
+	prepare()
+	test.isequal("/path/to/my/custom/clang", clang.gettoolname(cfg, "cc"))
+end
+
 --
```

Comment:
`toolsetpaths` is not tested.
How it behaves if we mix `toolsetpath` and `toolsetpaths`?
At which level the table are merged?